include/fastdds/dds/builtin/topic/ParticipantBuiltinTopicData.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file ParticipantBuiltinTopicData.hpp
+ *
+ */
+
+#ifndef PARTICIPANT_BUILTIN_TOPIC_DATA_HPP
+#define PARTICIPANT_BUILTIN_TOPIC_DATA_HPP
+
+#include <fastdds/dds/builtin/topic/BuiltinTopicKey.hpp>
+#include <fastdds/dds/core/policy/QosPolicies.hpp>
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+namespace builtin {
+
+struct ParticipantBuiltinTopicData
+{
+    //! Builtin topic Key
+    BuiltinTopicKey_t key;
+
+    //! UserData QoS
+    UserDataQosPolicy user_data;
+};
```

Comment:
Should we include our own extensions? Other vendors include things like `participant_name`, `vendor_id`, `protocol_version`, `builtin_endpoints`, `locators` and `properties`. 

On the old API, the equivalent to this structure is ParticipantProxyData, which is included as part of `ParticipantDiscoveryInfo`. As this structure is designed for system introspection, I think we should offer more than what the standard says.

That said, we may leave the extensions for the time when we implement the introspection methods.

Reply:
I think it is impossible to decide which variables these classes must have without a proper previous design of the introspection.
However, we can always extend the API afterwards.